core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -166,6 +169,13 @@ public PartitionSpec spec() {
   }
 
   public ManifestReader<F> filterRows(Expression expr) {
+    if (this.content.equals(FileType.DELETE_FILES) && this.metadataSchema.identifierFieldIds() != null &&
+        !this.metadataSchema.identifierFieldIds().isEmpty()) {
```

Comment:
Nit: normally we only use `this` when accessing properties if we’re assigning to them.

They’re not needed here.

Reply:
Note these style nits can be resolved when we’re closer to merging the PR if you’d like.